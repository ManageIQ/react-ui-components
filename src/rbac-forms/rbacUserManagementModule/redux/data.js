export const data = {
  rows: [
    {
      id: 100,
      name: 'Adam the first',
      userid: 'Admin',
      email: 'email@neco.com',
      current_group: {
        label: 'Some group',
        onClick: () => console.log('Current group clicked'),
      },
      role: {
        label: 'Admin role',
        onClick: () => console.log('Current role click'),
      },
      groups: [{
        label: 'Cloud-Operators',
        icon: 'group',
        groupId: 1,
        onClick: () => console.log('Group click'),
      }, {
        label: 'Cloud-Users',
        icon: 'group',
        groupId: 2,
        onClick: () => console.log('Group click'),
      }],
      lastlogon: '06/25/18 06:58:14 UTC',
      lastlogoff: '06/30/18 06:58:14 UTC',
    },
    {
      id: 101,
      name: 'Marge the middle',
      userid: 'La transparenza della qualita',
      email: 'email@jandak.com',
      current_group: {
        label: 'Mediocre group',
        onClick: () => console.log('Current group clicked'),
      },
      role: {
        label: 'Shot role',
        onClick: () => console.log('Current role click'),
      },
      groups: [{
        label: 'Cloud-Operators',
        icon: 'group',
        groupId: 1,
        onClick: () => console.log('Group click'),
      }, {
        label: 'Cloud-Users',
        icon: 'group',
        groupId: 2,
        onClick: () => console.log('Group click'),
      }],
      lastlogon: '01/25/18 06:58:14 UTC',
      lastlogoff: '06/30/18 06:58:14 UTC',
    },
    {
      id: 102,
      name: 'Zlatko the last',
      userid: 'Peligro',
      email: 'email@jandak.com',
      current_group: {
        label: 'Admin group',
        onClick: () => console.log('Current group clicked'),
      },
      role: {
        label: 'Cloud-Users',
        onClick: () => console.log('Current role click'),
      },
      groups: [{
        label: 'Cloud-Operators',
        icon: 'group',
        groupId: 1,
        onClick: () => console.log('Group click'),
      }, {
        label: 'Cloud-Users',
        icon: 'group',
        groupId: 2,
        onClick: () => console.log('Group click'),
      }],
      lastlogon: '08/25/18 06:58:14 UTC',
      lastlogoff: '010/30/18 06:58:14 UTC',
    },
  ],
  columns: [{
    property: 'name',
    label: 'Full Name',
  }, {
    property: 'userid',
    label: 'Username',
  }, {
    property: 'email',
    label: 'E-mail',
  }, {
    property: 'current_group',
    label: 'Current Group',
  }, {
    property: 'role',
    label: 'Role',
  }, {
    property: 'lastlogon',
    label: 'Last Logon',
  }, {
    property: 'lastlogoff',
    label: 'Last Logoff',
  }],
  selectedUsers: null,
  groups: [
    {
      value: 1,
      label: 'Hope',
    },
    {
      value: 2,
      label: 'Christa',
    },
    {
      value: 3,
      label: 'Tania',
    },
    {
      value: 4,
      label: 'Mcintyre',
    },
    {
      value: 5,
      label: 'Tonya',
    },
    {
      value: 6,
      label: 'Marcie',
    },
    {
      value: 7,
      label: 'Cara',
    },
    {
      value: 8,
      label: 'Becker',
    },
    {
      value: 9,
      label: 'Foley',
    },
    {
      value: 10,
      label: 'Perkins',
    },
  ],
};
