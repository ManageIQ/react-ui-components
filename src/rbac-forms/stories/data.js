import { action } from '@storybook/addon-actions';

export const groups = [
  {
    value: 1,
    label: 'Hope',
  },
  {
    value: 2,
    label: 'Christa',
  },
  {
    value: 3,
    label: 'Tania',
  },
  {
    value: 4,
    label: 'Mcintyre',
  },
  {
    value: 5,
    label: 'Tonya',
  },
  {
    value: 6,
    label: 'Marcie',
  },
  {
    value: 7,
    label: 'Cara',
  },
  {
    value: 8,
    label: 'Becker',
  },
  {
    value: 9,
    label: 'Foley',
  },
  {
    value: 10,
    label: 'Perkins',
  },
];

export const editedUser = {
  name: 'Users full name',
  userid: 'admin jiny',
  chosen_group: [5, 7],
};

export const usersTableColumns = [{
  property: 'fullname',
  label: 'Full Name',
}, {
  property: 'username',
  label: 'Username',
}, {
  property: 'email',
  label: 'E-mail',
}, {
  property: 'currentgroup',
  label: 'Current Group',
}, {
  property: 'role',
  label: 'Role',
}, {
  property: 'lastlogon',
  label: 'Last Logon',
}, {
  property: 'lastlogoff',
  label: 'Last Logoff',
}];

export const usersTableRows = [
  {
    id: 100,
    fullname: 'Adam the first',
    username: 'Admin',
    email: 'email@neco.com',
    currentgroup: 'Some group',
    role: 'Admin role',
    lastlogon: '06/25/18 06:58:14 UTC',
    lastlogoff: '06/30/18 06:58:14 UTC',
  },
  {
    id: 101,
    fullname: 'Marge the middle',
    username: 'La transparenza della qualita',
    email: 'email@jandak.com',
    currentgroup: 'Mediocre group',
    role: 'Shot role',
    lastlogon: '01/25/18 06:58:14 UTC',
    lastlogoff: '06/30/18 06:58:14 UTC',
  },
  {
    id: 102,
    fullname: 'Zlatko the last',
    username: 'Peligro',
    email: 'email@jandak.com',
    currentgroup: 'Super group',
    role: 'Shot role',
    lastlogon: '08/25/18 06:58:14 UTC',
    lastlogoff: '010/30/18 06:58:14 UTC',
  },
];

export const user = {
  name: 'Administrator',
  userid: 'Admin',
  email: 'mail@mail.com',
  current_group: {
    label: 'EvmGroup-super_administrator',
    onClick: () => action('Current group clicked'),
  },
  groups: [{
    label: 'Cloud-Operators',
    icon: 'group',
    groupId: 1,
    onClick: () => action('Group click'),
  }, {
    label: 'Cloud-Users',
    icon: 'group',
    groupId: 2,
    onClick: () => action('Group click'),
  }],
  role: {
    label: 'Cloud-Users',
    onClick: () => action('Current role click'),
  },
};
