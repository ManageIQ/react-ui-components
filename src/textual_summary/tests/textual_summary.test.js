import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { TextualSummary } from '../index';
import { summary1 } from '../data/textual_summary';

describe('TextualSummary', () => {
  it('renders just fine', () => {
    const group = shallow(<TextualSummary summary={summary1} onClick={() => null} />);
    expect(toJson(group)).toMatchSnapshot();
  });
});
