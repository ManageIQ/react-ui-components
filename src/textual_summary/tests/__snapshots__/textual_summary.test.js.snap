// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextualSummary renders just fine 1`] = `
<div
  className="row"
>
  <TextualRow
    groups={
      Array [
        Object {
          "component": "GenericGroup",
          "items": Array [
            Object {
              "hoverClass": "no-hover",
              "label": "Region",
              "value": "US East (N. Virginia)",
            },
            Object {
              "hoverClass": "no-hover",
              "label": "Type",
              "value": "Amazon EC2 Network",
            },
            Object {
              "hoverClass": "no-hover",
              "label": "Management Engine GUID",
              "value": null,
            },
          ],
          "title": "Properties",
        },
        Object {
          "component": "GenericGroup",
          "items": Array [
            Object {
              "hoverClass": "no-hover",
              "label": "Default Credentials",
              "title": "Ok",
              "value": "Valid",
            },
            Object {
              "hoverClass": "no-hover",
              "label": "SmartState Docker Credentials",
              "title": "Ok",
              "value": "Valid",
            },
            Object {
              "hoverClass": "no-hover",
              "label": "Last Refresh",
              "title": null,
              "value": Array [
                Object {
                  "value": "Success - 4 Months Ago",
                },
                Object {
                  "value": null,
                },
              ],
            },
            Object {
              "hoverClass": "no-hover",
              "label": "Last Refresh Date",
              "value": "2018-01-20T23:35:50.908Z",
            },
          ],
          "title": "Status",
        },
      ]
    }
    key="0"
    onClick={[Function]}
  />
  <TextualRow
    groups={
      Array [
        Object {
          "component": "GenericGroup",
          "items": Array [
            Object {
              "hoverClass": "no-hover",
              "icon": "pficon pficon-cloud-tenant",
              "image": null,
              "label": "Cloud Tenants",
              "value": "0",
            },
            Object {
              "hoverClass": "",
              "icon": "ff ff-cloud-network",
              "image": null,
              "label": "Cloud Networks",
              "link": "/ems_network/39?display=cloud_networks",
              "title": "Show all Cloud Networks",
              "value": "1",
            },
            Object {
              "hoverClass": "",
              "icon": "pficon pficon-network",
              "image": null,
              "label": "Cloud Subnets",
              "link": "/ems_network/39?display=cloud_subnets",
              "title": "Show all Cloud Subnets",
              "value": "6",
            },
            Object {
              "hoverClass": "",
              "icon": "pficon pficon-route",
              "image": null,
              "label": "Network Routers",
              "link": "/ems_network/39?display=network_routers",
              "title": "Show all Network Routers",
              "value": "1",
            },
            Object {
              "hoverClass": "",
              "icon": "pficon pficon-cloud-security",
              "image": null,
              "label": "Security Groups",
              "link": "/ems_network/39?display=security_groups",
              "title": "Show all Security Groups",
              "value": "18",
            },
            Object {
              "hoverClass": "",
              "icon": "ff ff-floating-ip",
              "image": null,
              "label": "Floating IPs",
              "link": "/ems_network/39?display=floating_ips",
              "title": "Show all Floating IPs",
              "value": "14",
            },
            Object {
              "hoverClass": "",
              "icon": "ff ff-network-port",
              "image": null,
              "label": "Network Ports",
              "link": "/ems_network/39?display=network_ports",
              "title": "Show all Network Ports",
              "value": "26",
            },
            Object {
              "hoverClass": "",
              "icon": "ff ff-load-balancer",
              "image": null,
              "label": "Load Balancers",
              "link": "/ems_network/39?display=load_balancers",
              "title": "Show all Load Balancers",
              "value": "2",
            },
            Object {
              "background": "green",
              "hoverClass": "",
              "icon": "fa fa-play",
              "image": null,
              "label": "Power State",
              "value": "On",
            },
          ],
          "title": "Relationships",
        },
        Object {
          "component": "GenericGroup",
          "items": Array [
            Object {
              "hoverClass": "",
              "icon": "pficon pficon-topology",
              "label": "Topology",
              "link": "/network_topology/show/39",
              "title": "Show topology",
            },
          ],
          "title": "Overview",
        },
        Object {
          "component": "TagGroup",
          "items": Array [
            Object {
              "hoverClass": "no-hover",
              "icon": "pficon pficon-zone",
              "label": "Managed by Zone",
              "value": "Amazon Zone",
            },
            Object {
              "hoverClass": "no-hover",
              "icon": "fa fa-tag",
              "label": "My Company Tags",
              "value": "No My Company Tags have been assigned",
            },
          ],
          "title": "Smart Management",
        },
      ]
    }
    key="1"
    onClick={[Function]}
  />
</div>
`;
